<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>697</width>
    <height>445</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MIDI file import</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="childrenCollapsible">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="horizontalLayoutWidget_5">
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="1" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="0" rowspan="2">
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Library properties</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="5" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLabel" name="bankmap_lbl">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Map all sounds to bank</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="bankmap_combo">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="bankmap_btn">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Apply</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="3" column="0">
           <widget class="QTableWidget" name="sounds_table">
            <property name="contextMenuPolicy">
             <enum>Qt::CustomContextMenu</enum>
            </property>
            <property name="editTriggers">
             <set>QAbstractItemView::NoEditTriggers</set>
            </property>
            <property name="dragDropOverwriteMode">
             <bool>false</bool>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="label">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Source:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="GrowingFileLabel" name="source_lbl">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="file_open_btn">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Import MIDI file...</string>
              </property>
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="4" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <property name="spacing">
             <number>8</number>
            </property>
            <item>
             <widget class="QPushButton" name="select_all_btn">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Select all</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="check_btn">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Check selected sounds</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="uncheck_btn">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Uncheck selected sounds</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QCheckBox" name="doubleclick_dump_chk">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Dump to Sound Mode Edit Buffer on double click</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="editor_activate_chk">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>open in Sound editor</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QGroupBox" name="sound_groupbox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>250</width>
           <height>0</height>
          </size>
         </property>
         <property name="title">
          <string>Sound properties</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout">
            <item row="0" column="0" colspan="2">
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Name:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="name_edit">
                <property name="maxLength">
                 <number>16</number>
                </property>
                <property name="readOnly">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="cat_combo"/>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Category:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0" colspan="2">
             <widget class="QPushButton" name="apply_btn">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Apply</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="7" column="0">
           <widget class="QPushButton" name="single_dump_btn">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Dump now</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="Line" name="line_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="1" column="0">
             <widget class="QRadioButton" name="sound_buffer_radio">
              <property name="text">
               <string>Sound Buffer</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">destination_group</string>
              </attribute>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Dump destination:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QComboBox" name="bank_combo">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>48</width>
                <height>16777215</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="2" column="2">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Prog:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QRadioButton" name="multi_buffer_radio">
                <property name="text">
                 <string>Multi Buffer</string>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">destination_group</string>
                </attribute>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="multi_spin">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>16</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="2">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Bank:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2" colspan="2">
             <widget class="QRadioButton" name="library_radio">
              <property name="text">
               <string>Library</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">destination_group</string>
              </attribute>
             </widget>
            </item>
            <item row="2" column="3">
             <widget class="QSpinBox" name="prog_spin">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>128</number>
              </property>
             </widget>
            </item>
            <item row="0" column="1" rowspan="3">
             <widget class="Line" name="line">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Single sound dump</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Sound summary</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="SummaryWidget" name="summary_widget" native="true"/>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel</set>
     </property>
    </widget>
   </item>
  </layout>
  <zorder>buttonBox</zorder>
  <zorder>splitter</zorder>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SummaryWidget</class>
   <extends>QWidget</extends>
   <header>bigglesworth.dialogs.summary</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GrowingFileLabel</class>
   <extends>QLabel</extends>
   <header>bigglesworth.dialogs.midi_import</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Dialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>418</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>310</x>
     <y>418</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>select_all_btn</sender>
   <signal>clicked()</signal>
   <receiver>sounds_table</receiver>
   <slot>selectAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>309</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>multi_buffer_radio</sender>
   <signal>toggled(bool)</signal>
   <receiver>multi_spin</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>467</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>library_radio</sender>
   <signal>toggled(bool)</signal>
   <receiver>bank_combo</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>615</x>
     <y>129</y>
    </hint>
    <hint type="destinationlabel">
     <x>633</x>
     <y>154</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>library_radio</sender>
   <signal>toggled(bool)</signal>
   <receiver>prog_spin</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>615</x>
     <y>129</y>
    </hint>
    <hint type="destinationlabel">
     <x>633</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleclick_dump_chk</sender>
   <signal>toggled(bool)</signal>
   <receiver>editor_activate_chk</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>355</x>
     <y>68</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="destination_group"/>
 </buttongroups>
</ui>
